### PR TITLE
Add emacs SPEC to Mariner

### PR DESCRIPTION
**Pull request by chalamalasetty**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
Add emacs SPEC to Mariner

###### Change Log  

- Change
   Add emacs SPEC to Mariner

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Pipeline build id: Local build

### CODE REVIEW

File: SPECS/emacs/emacs.spec [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            --without-harfbuzz       \
+            --without-m17n-flt       \
+            --without-toolkit-scroll-bars \
+            --without-xaw3d          \
+            --without-xim            \
+            --without-makeinfo
+make %{?_smp_mflags}
+
+%install
+make DESTDIR=%{buildroot} %{?_smp_mflags} install
```
Comment (oliviacrain) [inline SPECS/emacs/emacs.spec:46]: Nit: Can use macros here
```suggestion
%make_install
```

File: SPECS/emacs/emacs.spec [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            --without-rsvg           \
+            --without-lcms2          \
+            --without-xft            \
+            --without-harfbuzz       \
+            --without-m17n-flt       \
+            --without-toolkit-scroll-bars \
+            --without-xaw3d          \
+            --without-xim            \
+            --without-makeinfo
+make %{?_smp_mflags}
```
Comment (oliviacrain) [inline SPECS/emacs/emacs.spec:43]: Nit: Can use macros here
```suggestion
%make_build
```

File: SPECS/emacs/emacs.spec [hunk 3]
```diff
@@ -0,0 +1,86 @@
+Summary:        GNU Emacs text editor
+Name:           emacs
+Version:        28.1
+Release:        2%{?dist}
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+License:        GPLv3+ and CC0-1.0
+URL:            http://www.gnu.org/software/emacs/
+Group:          Applications/Editors
+Source0:        https://ftp.gnu.org/gnu/emacs/emacs-%{version}.tar.xz
```
Comment (oliviacrain) [inline SPECS/emacs/emacs.spec:10]: Nit: Could use macros for the package name

```suggestion
Source0:        https://ftp.gnu.org/gnu/%{name}/%{name}-%{version}.tar.xz 
```

**Review by oliviacrain — APPROVED** [review]
Approved with suggestions

File: SPECS/emacs/emacs.spec [hunk 4]
```diff
@@ -0,0 +1,86 @@
+Summary:        GNU Emacs text editor
+Name:           emacs
+Version:        28.1
+Release:        2%{?dist}
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+License:        GPLv3+ and CC0-1.0
+URL:            http://www.gnu.org/software/emacs/
```
Comment (oliviacrain) [inline SPECS/emacs/emacs.spec:8]: Nit: Could be HTTPS

```suggestion
URL:            https://www.gnu.org/software/emacs/
```

File: SPECS/emacs/emacs.spec (hunk 1 above)
Comment (chalamalasetty) [reply SPECS/emacs/emacs.spec]: updated.

File: SPECS/emacs/emacs.spec (hunk 2 above)
Comment (chalamalasetty) [reply SPECS/emacs/emacs.spec]: updated.

File: SPECS/emacs/emacs.spec (hunk 3 above)
Comment (chalamalasetty) [reply SPECS/emacs/emacs.spec]: updated

File: SPECS/emacs/emacs.spec (hunk 4 above)
Comment (chalamalasetty) [reply SPECS/emacs/emacs.spec]: updated.

**Review by oliviacrain — APPROVED** [review]

**Review by neha170 — APPROVED** [review]

**Pull request merged**